### PR TITLE
Change java metadata['java_version'] to parsed JvmVersion

### PR DESCRIPTION
**Pull request by uriariel**
Change java metadata['java_version'] to parsed JvmVersion

## Description
Change java metadata['java_version'] to parsed JvmVersion

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the relevant documentation.
- [x] I have added tests for new logic.

### CODE REVIEW

**Comment by Jongy** [comment]
The rationale in @uriariel 's change is to allow easy access to parsed `JvmVersion`s in our backend.

What makes more sense instead is to move the parsing logic to granulate-utils which makes it usable by other components:
1. To avoid changing the protocol format (existing users of `java_version` in the metadata, and the need to accommodate old gProfiler versions which still send it the old way)
2. `parse_jvm_version` has changed multiple times. If we keep gProfiler sending the raw data, then once we update the code we have the most recent parsing available (instead of storing parsed data, which means we could store data that was parsed incorrectly, in case we fix the code).

TL;DR @uriariel we will move the code to granulate-utils if we need access to `JvmVersion`.

**Comment by Jongy** [comment]
Superseded by https://github.com/Granulate/granulate-utils/pull/65
